runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -138,7 +138,7 @@ public synchronized void acquireBuffer() {
     public int getSizeEstimate() {
         byte[] tempData = data;
         if (tempData != null) {
-            return tempData.length;
+            return tempData.length + metadataMap.size() + DataType.values().length;
```

Comment:
metadataMap.size() is the number of entries, not the size.

Reply:
i could not find a straightforward way to calculate the number of bytes used by metadataMap.  size() is not supported on the key and value types.  Similary for DataType - had to use the number of data types in the enum.  

I can get the size from the byteBuffer in the constructor.